### DISCUSSION TITLE
Container horizontal padding

### DISCUSSION THREAD

**Discussion by JPortegijs** [tailwindlabs/tailwindcss#9841] [Help] [answered]
Hello,

I'm trying to apply horizontal padding on a container at a custom breakpoint (xs), but it doesn't seem to be applied. Anything I'm doing incorrectly?

```javascript
theme: {
    screens: {
      xs: '375px',
      ...defaultTheme.screens,
    },

    container: (theme) => {
      return {
        center: true,
        padding: {
          DEFAULT: '.75rem',
          xs: '2rem',
          sm: '2rem',
          md: '2rem',
          lg: '4rem',
          xl: '5rem',
          '2xl': '6rem',
        },
        screens: {
          ...theme('screens'),
          xs: '100%',
        },
      };
    },
  }
```

**Comment by wongjn** [comment, marked as answer]
Consider removing the `screens` object within the `container` function. The `100%` value for `xs` means the corresponding `padding` won't ever activate since `@media (min-width: 100%)` never applies. You may have been intending this to affect the `max-width` declaration only, but it also affects the media query. You might be better off using your own container component out of Tailwind utilities instead. Also, you'd want to use the deconstructed `theme` via `({ theme }) => {}`, not the full argument itself (`(theme) => {}`).

**Comment by JPortegijs** [comment]
@wongjn Thanks! I fixed it by writing my own CSS like this:
```css
@layer utilities {
  @media (min-width: 375px) and (max-width: 639px) {
    .container {
      width: 100%;
      max-width: none;
    }
  }
}
```
